doc/introduction/data.rst

```diff
@@ -161,7 +170,7 @@ Quantum Datasets Functions and Classes
 .. autosummary::
     :nosignatures:
 
-    ~pennylane.data.list_datasets
```

Comment:
Since we getting rid of a public facing function without a deprecation cycle, this might be a breaking change that's worth a mention in the changelog.

Reply:
Hey @astralcai, thanks for taking a look at this! We're in the process of deciding what to do here. It's a fairly involved task to keep this function working with the new backend. This may come in a separate PR, but we'll keep everyone updated here.